internal/db/query.go

```diff
@@ -274,6 +275,58 @@ func GetTars(ctx context.Context, tx pgx.Tx, project int64, vrange VersionRange,
 	}, nil
 }
 
+func latestCacheTars() string {
+	return `
+		WITH latest_cache_version AS (
+			SELECT version, hashes
+			FROM dl.cache_versions
+			ORDER BY version DESC LIMIT 1
+		),
+		version_hash AS (
+			SELECT version, unnest(hashes) AS hash
+			FROM latest_cache_version
+		)
+		SELECT version, version_hash.hash, bytes
+		FROM version_hash
+		JOIN dl.contents
+		ON version_hash.hash = dl.contents.hash
+	`
+}
+
+type cacheTarStream func() (int64, []byte, *string, error)
+
+func GetCacheTars(ctx context.Context, tx pgx.Tx) (cacheTarStream, error) {
+	rows, err := tx.Query(ctx, latestCacheTars())
+	if err != nil {
+		return nil, fmt.Errorf("GetCachedTars query: %w", err)
+	}
+
+	var version int64
+
+	stream := func() (int64, []byte, *string, error) {
+		if !rows.Next() {
+			return 0, nil, nil, io.EOF
+		}
+		var rowVersion int64
+		var hash string
+		var encoded []byte
+
+		err := rows.Scan(&rowVersion, &hash, &encoded)
```

Comment:
If you pull the latest from main we now have a `Hash` type for these instead of just `[]byte`.

And to read it easily from PGX (there are a few examples hanging around) you can do something like:

```
SELECT (hash).h1, (hash).h2
```

```
var hash Hash
rows.Scan(&hash.H1, &hash.H2)
```